nncf/quantization/algorithms/weight_compression/openvino_backend.py

```diff
@@ -153,51 +153,37 @@ def transform_model(
             compressed_const = opset.constant(
                 compressed_weight.tensor.data, dtype=compression_dtype, name=const_node_name
             )
-            converted_const = opset.convert(compressed_const, ov.Type.f32)
+            converted_const = opset.convert(compressed_const, ov.Type.f16)
             if compressed_weight.zero_point is not None:
                 zero_point_const = opset.constant(
                     compressed_weight.zero_point.data,
                     dtype=compression_dtype,
                     name=f"{const_node_name}/zero_point",
                 )
-                converted_zero_point = opset.convert(zero_point_const, ov.Type.f32)
+                converted_zero_point = opset.convert(zero_point_const, ov.Type.f16)
                 converted_const = opset.subtract(
                     converted_const, converted_zero_point, name=f"{const_node_name}/zero_point/subtract"
                 )
 
             scale_const = opset.constant(
                 compressed_weight.scale.data, dtype=ov.Type.f16, name=f"{const_node_name}/scale"
             )
-            scale_const = opset.convert(scale_const, ov.Type.f32, name=f"{const_node_name}/scale_convert")
             mul = opset.multiply(
                 converted_const,
                 scale_const,
                 name=f"{const_node_name}/fq_weights_{wc_params.weight_port_id}",
             )
+            if const_dtype == ov.Type.f32:
+                mul = opset.convert(
+                    mul, ov.Type.f32, name=f"{mul.get_friendly_name()}/convert"
+                )
```

Comment:
It looks like this code supports only fp32 and fp16. What will happen to other types?
```suggestion
            if const_dtype != ov.Type.f16:
                mul = opset.convert(
                    mul, const_dtype, name=f"{mul.get_friendly_name()}/convert"
                )
```

Reply:
Thanks, changed.